src/zap_hooks/soos_zap_hook.py

```diff
@@ -77,12 +77,16 @@ def zap_pre_shutdown(zap):
         serialize_and_save(zap.core, 'wrk/core_data_pre_shutdown.json')
         serialize_and_save(zap.pscan, 'wrk/pscan_data_pre_shutdown.json')
         serialize_and_save(zap.context, 'wrk/context_data_pre_shutdown.json')
-    log("Overview of spidered URL's:")
+    log("Spidered URLs:")
     with open('spidered_urls.txt', 'w') as f:
         for url in zap.spider.all_urls:
             f.write(f"{url}\n")
-            log(f"found: {url}")
-
+            log(f"spider discovered url: {url}")
+    log("Core URLs:")
+    with open('core_urls.txt', 'w') as f:
+        for url in zap.core.urls():
```

Comment:
this is what the API scan uses to dump out the url count - https://github.com/zaproxy/zaproxy/blob/main/docker/zap-api-scan.py

Reply:
logging.info('Number of Imported URLs: ' + str(len(urls)))

and

print('Total of ' + str(num_urls) + ' URLs')

(which are different before and after scan